test/run-tests

```diff
@@ -556,38 +527,39 @@ def choose_task(gh, repos, images, config, ansible_id, args):
     """
 
     for owner, repo in repos:
-        pulls = gh.get(f"repos/{owner}/{repo}/pulls").json()
+        pyghrepo = gh.get_repo(f"{owner}/{repo}")
+        pulls = [pr for pr in pyghrepo.get_pulls(state="open")]
         random.shuffle(pulls)
         for pull in pulls:
-            logging.debug("Evaluating PR {}".format(pull["url"]))
-            author = pull["user"]["login"]
-            head = pull["head"]["sha"]
-            number = pull["number"]
+            logging.debug(f"Evaluating PR pull {pull.url}")
+            author = pull.user.login
+            head = pull.head.sha
+            number = pull.number
 
-            if not pull_ok_to_test(gh, owner, repo, number, author, head):
+            if not pull_ok_to_test(pyghrepo, pull, author, head):
                 continue
 
-            statuses = get_statuses(gh, owner, repo, head, args.max_num_statuses)
-            commands = get_comment_commands(gh, owner, repo, number)
+            statuses = get_statuses(pull.get_commits())
```

Comment:
to get the list of statuses, you need to get the commit using e.g. `commit = repo.get_commit(pull.head.sha)`, then use `commit.get_combined_status().statuses` to get the list of statuses.